doc/W-ECMP/W-ECMP-test-plan.md

```diff
@@ -0,0 +1,196 @@
+
+## Overview
+In ECMP (Equal-Cost Multi-Path) scenario, traffic is distributed equally across all available paths and each path has same cost.
+
+In W-ECMP (Weighted-Cost Multi-Path) scenario, it can distribute traffic base on the weight value assigned to each path. This is useful when the paths have different bandwidth capacities. In a W-ECMP setup, each path is assigned a weight based on various criteria, such as bandwidth, delay, or administrative preferences. BGP then uses these weights to distribute outbound traffic among the multiple available paths
+
+
+## BGP link bandwidth extended community
+The BGP link bandwidth extended community contains information about the bandwidth of a link. This information is advertised along with BGP route updates to inform other routers about the capacity of the link.
+RFC: [https://datatracker.ietf.org/doc/html/draft-ietf-idr-link-bandwidth]
+
+```
+router bgp 65200
+ bgp router-id 10.1.0.2
+ no bgp ebgp-requires-policy
+ neighbor SPINE1 peer-group
+ neighbor SPINE1 remote-as 65100
+ neighbor 10.10.1.1 peer-group SPINE1
+ neighbor 10.10.2.1 peer-group SPINE1
+ !
+ address-family ipv4 unicast
+  network 20.20.20.0/24
+  neighbor SPINE1 route-map TO_BGP_PEER_V4 out      <<<<<<
+ exit-address-family
+exit
+!
+route-map TO_BGP_PEER_V4 permit 100
+ set extcommunity bandwidth num-multipaths          <<<<<<
+exit
+!
+route-map TO_BGP_PEER_V6 permit 100
+ set extcommunity bandwidth num-multipaths
+exit
+```
+
+### Disable W-ECMP
+If W-ECMP is `disabled`, the BGP update message contains three default path attributes `ORIGIN`/`AS_PATH`/`NEXT_HOP` when advertising network prefix 10.0.0.0/24
+![Use case scenario](images/W-ECMP_disabled.png)
+
+
+### Enable W-ECMP
+If W-ECMP is `enabled`, the BGP `EXTENDED_COMMUNITIES` attribute is added in the update message (yellow part)
+![Use case scenario](images/W-ECMP_enabled.png)
+
+
+## Related DUT CLI commands
+
+### Config
+The following command can be used to enable/disable W-ECMP:
+```
+config bgp device-global wcmp enabled
+config bgp device-global wcmp disabled
+```
+
+### Show
+The following command can be used to show W-ECMP status:
+```
+show bgp device-global
+show bgp device-global --json
+```
+Example:
+```
+admin@router:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  enabled
+```
+```
+admin@router:~$ show bgp device-global --json
+{
+    "tsa": "disabled",
+    "wcmp": "enabled"
+}
+```
+
+
+## Related DUT configuration files
+```
+"BGP_DEVICE_GLOBAL": {
+    "STATE": {
+        "tsa_enabled": "false",
+        "wcmp_enabled": "true"
+    }
+}
+```
+
+
+## Test cases
+### Test case #1 - W-ECMP command line test
+1. Enable W-ECMP and verify W-ECMP status
+```
+admin@router:~$ sudo config bgp device-global wcmp enabled
+admin@router:~$ show bgp device-global
+TSA       WCMP
+--------  -------
+disabled  enabled
+```
+2. Verify the route weight value on remote host is 25, which means W-ECMP is effect
```

Comment:
@weiguo-nvidia FYI. Need to update the test plan, once the FRR normalization to 255 is backported